pkg/operator/configobserver/cloudprovider/observe_cloudprovider.go

```diff
@@ -66,7 +68,25 @@ func (c *cloudProviderObserver) ObserveCloudProviderNames(genericListers configo
 		return existingConfig, append(errs, err)
 	}
 
+	featureGate, err := listers.FeatureGateLister().Get("cluster")
+	if errors.IsNotFound(err) {
+		recorder.Eventf("ObserveCloudProviderNames", "Optional featuregate.%s/cluster not found", configv1.GroupName)
+	} else if err != nil {
+		return existingConfig, append(errs, err)
+	}
+
+	external, err := cloudprovider.IsCloudProviderExternal(infrastructure.Status.Platform, featureGate)
+	if err != nil {
+		recorder.Eventf("ObserveCloudProviderNames", "Could not determine external cloud provider state: %v", err)
```

Comment:
why wouldn't we return the previous config when hitting this condition?

Reply:
This just means that the `FeatureGate` resource is formatted incorrectly, build usual cloud-config.

Reply:
> This just means that the `FeatureGate` resource is formatted incorrectly, build usual cloud-config.

You still want to return the previous config as opposed to blanking it out.